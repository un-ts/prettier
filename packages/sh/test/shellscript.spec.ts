import prettier from 'prettier'
import { ParseError } from 'sh-syntax'

import ShPlugin, { type IShParseError } from 'prettier-plugin-sh'

test('fatal parse error with meaningful message', () => {
  try {
    prettier.format(`echo )`, {
      filepath: 'broken.sh',
      parser: 'sh',
      plugins: [ShPlugin],
      pluginSearchDirs: false,
    })
  } catch (err: unknown) {
    expect(((err as Error).cause as IShParseError).Text).toMatchSnapshot()
  }

  try {
    prettier.format(`echo )`, {
      filepath: 'broken.sh',
      parser: 'sh',
      plugins: [ShPlugin],
      pluginSearchDirs: false,
      // @ts-expect-error
      experimentalWasm: true,
    })
  } catch (err: unknown) {
    const error = (err as Error).cause as ParseError | undefined
    expect(error?.Text || error?.message).toMatchSnapshot()
  }
})
