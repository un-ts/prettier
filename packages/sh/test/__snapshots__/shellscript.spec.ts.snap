// Vitest Snapshot v1

exports[`fatal parse error with meaningful message 1`] = `"a command can only contain words and redirects; encountered )"`;

exports[`fatal parse error with meaningful message 2`] = `"a command can only contain words and redirects; encountered )"`;
